pkg/promlib/models/query.go

```diff
@@ -188,12 +191,19 @@ func Parse(query backend.DataQuery, dsScrapeInterval string, intervalCalculator
 		dsScrapeInterval,
 		timeRange,
 	)
-	if enableScope && model.Scope != nil && len(model.Scope.Filters) > 0 {
-		expr, err = ApplyQueryScope(expr, *model.Scope)
+
+	if enableScope {
+		var scopeFilters []ScopeFilter
+		if model.Scope != nil {
+			scopeFilters = model.Scope.Filters
```

Comment:
Maybe check if Filters is also not null just to be sure.

Reply:
ApplyQueryFilters handles either slice being nil, in `scope_test.go` you can see that some test cases have them has nil input.